Xamarin.Forms.Core/Storyboard.cs

```diff
@@ -0,0 +1,551 @@
+				e.PropertyName == PositionAnimation.ToProperty.PropertyName)
+			{
+				Begin(sender as Timeline);
+			}
+		}
+
+		IEnumerator IEnumerable.GetEnumerator()
+		{
+			return _timelines.GetEnumerator();
+		}
+
+		public int IndexOf(Timeline item)
+		{
+			return Children.IndexOf(item);
+		}
+
+		public void Insert(int index, Timeline item)
+		{
+			Children.Insert(index, item);
+		}
+
+		public void Add(Timeline item)
+		{
+			Children.Add(item);
+			item.PropertyChanged += AnimationPropertyChanged;
+
+			//((IList<IAnimation>)Children).Add(item);
+		}
+
+		public bool Contains(Timeline item)
+		{
+			return Children.Contains(item);
+		}
+
+		public void CopyTo(Timeline[] array, int arrayIndex)
+		{
+			Children.CopyTo(array, arrayIndex);
+		}
+
+		public bool Remove(Timeline item)
+		{
+			item.PropertyChanged -= AnimationPropertyChanged;
+
+			return Children.Remove(item);
+		}
+
+		IEnumerator<Timeline> IEnumerable<Timeline>.GetEnumerator()
+		{
+			return Children.GetEnumerator();
+		}
+
+		public void RemoveAt(int index)
+		{
+			Children.RemoveAt(index);
+		}
+
+		public void Clear()
+		{
+			Children.Clear();
+		}
+
+		//Storyboard.SetTarget(animOpacity, Control_);
+		//  Storyboard.SetTarget(animTransform, Control_);
+		//  Storyboard.SetTargetProperty(animOpacity, new PropertyPath("Opacity"));
+		//  Storyboard.SetTargetProperty(animTransform, new PropertyPath(
+
+		//public void Pause();
+		//public void Resume();
+		//public static void SetTarget(this Storyboard s, string target);
+		//public static void SetTargetProperty(this Storyboard s, string property);
+	}
+
+	public interface IInterpolatable
+	{
+		object InterpolateTo(object from, object target, double interpolation);
+	}
+
+	public interface IAnimation
+	{
+		void Begin();
+	}
+
+	public interface IAnimation<T> : IAnimation
+	{
+		T From { get; set; }
+		T To { get; set; }
+	}
+
+	public interface IInterpolatable<T> : IInterpolatable
+	{
+		T InterpolateTo(T from, T target, double interpolation);
+	}
+
+	public abstract class Animation<T> : Timeline
+	{
+		public abstract T From { get; set; }
+		public abstract T To { get; set; }
+	}
+
+
+	public class AnimateExtension : BindableObject, IMarkupExtension<Timeline>
+	{
+
+		public BindableProperty Property { set; get; }
+
+		public string Binding { set; get; }
+
+		public Timeline ProvideValue(IServiceProvider serviceProvider)
+		{
+			var anim = new PositionAnimation();
+
+			BindingBase GetBinding()
+			{
+				return new Binding(Binding);
+			}
+
+			anim.SetBinding(PositionAnimation.ToProperty, GetBinding());
+			Storyboard.SetTargetProperty(anim, Property);
+			return anim;
+		}
+
+		object IMarkupExtension.ProvideValue(IServiceProvider serviceProvider)
+		{
+			return (this as IMarkupExtension<Timeline>).ProvideValue(serviceProvider);
+		}
+	}
+
+
+	public abstract class InterpolatableAnimation<T> : Animation<T> where T : IInterpolatable<T>
+	{
+	}
+
+
+	public class PositionAnimation : Animation<int>
```

Comment:
Why is it called PositionAnimation? One example shows animating the Y property:

<PositionAnimation To="{Binding Y}" Storyboard.TargetProperty="Frame.Y" />
but the Y property is a double and the PositionAnimation is implemented with int:

public class PositionAnimation : Animation<int>
I'm confused, why's it called "PositionAnimation"?

Reply:
Why can't I just use DoubleAnimation ?

Reply:
In this case, it makes sense to have an `IntAnimation` to animate int type properties. For example, animate the FontSize of a Label.